src/Elastic.Apm/Logging/IApmLoggingExtensions.cs

```diff
@@ -3,157 +3,179 @@
 // See the LICENSE file in the project root for more information
 
 using System;
-using System.Collections.Concurrent;
+using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
 using System.Runtime.CompilerServices;
 using Elastic.Apm.Helpers;
 
-namespace Elastic.Apm.Logging
+namespace Elastic.Apm.Logging;
+
+internal static class LoggingExtensions
 {
-	internal static class LoggingExtensions
+	private static ConditionalWeakTable<string, LogValuesFormatter> Formatters { get; } = new();
+
+	internal static ScopedLogger Scoped(this IApmLogger logger, string scope) => new(logger is ScopedLogger s ? s.Logger : logger, scope);
+
+	private static void DoLog(this IApmLogger logger, LogLevel level, string message, Exception e, params object[] args)
 	{
-		private static ConcurrentDictionary<string, LogValuesFormatter> Formatters { get; } = new ConcurrentDictionary<string, LogValuesFormatter>();
```

Comment:
As a reviewing note, the .NET runtime implements this with a `ConcurrentDictionary` and protects it by dropping anything if more than N items are cached. For now this seems like a good middle ground though.